lib/spaceship/launcher.rb

```diff
@@ -88,6 +88,9 @@ def certificate
 
     # @return (Class) Access the provisioning profiles for this spaceship
     def provisioning_profile
+      app
+      device
+      certificate
       Spaceship::ProvisioningProfile.set_client(@client)
```

Comment:
ProvisioningProfile is the only class that eager-loads the other objects so the others are fine.  

I did consider having the `login` method go through all of the other classes and call set_client right away but I assumed there was a reason you were waiting to do this the individual methods. I can make that change if you think it's better/more clear/more future proof.